### PR TITLE
PYTHON-1834 (cont) Add pre-commit config

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  hooks:
+  - id: check-added-large-files
+  - id: check-case-conflict
+  - id: check-toml
+  - id: check-yaml
+  - id: debug-statements
+  - id: end-of-file-fixer
+    exclude: WHEEL
+    exclude_types: [json]
+  - id: forbid-new-submodules
```
Comment (behackett) [inline .pre-commit-config.yaml:14]: I don't see this in the pre-commit docs. I can guess what it does, but could be wrong. What effect will this have on our workflow?

Comment (blink1073) [reply .pre-commit-config.yaml:14]: It "prevents addition of new git submodules", should have no effect other than preventing a foot-gun.

Comment (blink1073) [reply .pre-commit-config.yaml:14]: Here's the full list of hooks docs: https://pre-commit.com/hooks.html

Comment (behackett) [reply .pre-commit-config.yaml:14]: Ah, I was looking at the wrong docs. Thanks.

**Review by behackett — APPROVED** [review]

**Review by juliusgeo — COMMENTED** [review]
LGTM!

**Review by juliusgeo — APPROVED** [review]

**Pull request merged**